lib/new_relic/agent/commands/agent_command_router.rb

```diff
@@ -29,7 +29,9 @@ def initialize(event_listener = nil)
           @handlers['start_profiler'] = proc { |cmd| thread_profiler_session.handle_start_command(cmd) }
           @handlers['stop_profiler'] = proc { |cmd| thread_profiler_session.handle_stop_command(cmd) }
 
-          if event_listener
+          event_listener&.subscribe(:before_shutdown, &method(:on_before_shutdown))
+          # the following statement needs else branch coverage
+          if event_listener # rubocop:disable Style/SafeNavigation
             event_listener.subscribe(:before_shutdown, &method(:on_before_shutdown))
```

Comment:
I think it's doing it twice now (line 32 and also 35) instead of once like before